### ISSUE TITLE
Field names

### ISSUE THREAD

**Issue by pappastech** [partkeepr/PartKeepr#425]
Is there a way to change the field names and add more fields to the part manager in Partkeepr? The field names currently in place don't reflect some common conventions for naming unless I'm missing something. For example, the 'Name' field in the part manager appears to be used as a part number field; the title of this field should then be Part Number.

Maybe I'm just missing the part number field...if I am, can someone please point me to where this field exists?

Thanks.

**Comment by felicitus** [comment]
The part name field is what it says it is - why do you think this is the part number? Also, what exactly do you define as part number?

**Comment by felicitus** [comment]
I just did a quick research, it seems that in Germany, we use "part name" instead of "part number", so that explains the confusion. However, I'd still prefer to leave it as name, because PartKeepr can be used for non-electronic components as well.

**Comment by pappastech** [comment]
How difficult would it be to make the field names configurable or make it possible to add/remove fields?

----- Reply message -----

**Comment by pappastech** [comment]
I'm of course willing to make the edits myself, but a quick grep of the source code reveals that all these field names are hard coded and in more than one file. Am I correct here?

**Comment by felicitus** [comment]
Yes, that's the case. What you could do is to rename the field in the frontend only.

**Comment by felicitus** [comment]
Configurable fields is an idea, but I'm not sure yet on how to implement this in a proper way.

**Comment by pappastech** [comment]
I'll take a crack and coming up with at least a configurable field name implementation. If I'm happy with the code, I'll send it back to the project for incorporation if you want to use it.

Is there a flowchart that would show me all the places that these field names are defined by chance? I can find them all by searching, it's just time consuming.

Thanks.

**Comment by felicitus** [comment]
Unfortunately there's no flowchart to help you with finding all occurrences of the field name. 

However, I'm considering a rename of the field or introduce an additional field for that purpose.

**Comment by felicitus** [comment]
This is indeed an issue. A linking between the technical field name and the canonical field name must be implemented at some point.

As it can seen in https://github.com/partkeepr/PartKeepr/issues/435#issuecomment-147776358 the technical field names in the exporter might confuse users.

If a configurable mapping between technical field name and canonical name is implemented, it would be useful to give administrators a way to re-configure the mapping.
